flink/v1.13/flink/src/test/java/org/apache/iceberg/flink/source/TestProjectMetaColumn.java

```diff
@@ -0,0 +1,180 @@
+import org.apache.flink.table.data.RowData;
+import org.apache.iceberg.DataFile;
+import org.apache.iceberg.DeleteFile;
+import org.apache.iceberg.FileFormat;
+import org.apache.iceberg.RowDelta;
+import org.apache.iceberg.SerializableTable;
+import org.apache.iceberg.Table;
+import org.apache.iceberg.TableProperties;
+import org.apache.iceberg.flink.SimpleDataUtil;
+import org.apache.iceberg.flink.TableLoader;
+import org.apache.iceberg.flink.TestHelpers;
+import org.apache.iceberg.flink.data.RowDataProjection;
+import org.apache.iceberg.flink.sink.RowDataTaskWriterFactory;
+import org.apache.iceberg.flink.sink.TaskWriterFactory;
+import org.apache.iceberg.io.TaskWriter;
+import org.apache.iceberg.io.WriteResult;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableMap;
+import org.apache.iceberg.relocated.com.google.common.collect.Lists;
+import org.junit.Assert;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.rules.TemporaryFolder;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestProjectMetaColumn {
+
+  @Rule
+  public final TemporaryFolder folder = new TemporaryFolder();
+  private final FileFormat format;
+
+  @Parameterized.Parameters(name = "fileFormat={0}")
+  public static Iterable<Object[]> parameters() {
+    return Lists.newArrayList(
+        new Object[] {FileFormat.PARQUET},
+        new Object[] {FileFormat.ORC},
+        new Object[] {FileFormat.AVRO}
+    );
+  }
+
+  public TestProjectMetaColumn(FileFormat format) {
+    this.format = format;
+  }
+
+  private void testSkipToRemoveMetaColumn(int formatVersion) throws IOException {
+    // Create the table with given format version.
+    String location = folder.getRoot().getAbsolutePath();
+    Table table = SimpleDataUtil.createTable(location,
+        ImmutableMap.of("format-version", String.valueOf(formatVersion)),
+        false);
+
+    List<RowData> rows = Lists.newArrayList(
+        SimpleDataUtil.createInsert(1, "AAA"),
+        SimpleDataUtil.createInsert(2, "BBB"),
+        SimpleDataUtil.createInsert(3, "CCC")
+    );
+    writeAndCommit(table, ImmutableList.of(), false, rows);
+
+    FlinkInputFormat input = FlinkSource
+        .forRowData()
+        .tableLoader(TableLoader.fromHadoopTable(location))
+        .buildFormat();
+
+    List<RowData> results = Lists.newArrayList();
+    TestHelpers.readRowData(input, rowData -> {
+      // If project to remove the meta columns, it will get a RowDataProjection.
+      Assert.assertTrue(rowData instanceof GenericRowData);
+      results.add(TestHelpers.copyRowData(rowData, SimpleDataUtil.ROW_TYPE));
+    });
+
+    // Assert the results.
+    TestHelpers.assertRows(rows, results, SimpleDataUtil.ROW_TYPE);
+  }
+
+  @Test
+  public void testV1SkipToRemoveMetaColumn() throws IOException {
+    testSkipToRemoveMetaColumn(1);
```

Comment:
why don't we move formatVersion to test parameters too?

Reply:
Thanks @stevenzwu for the comment. I did not use the parameterized `formatVersion` approach before because not all of the unit tests are need to be addressed in both v1 and v2 tables.  For example,  the `testV2RemoveMetaColumn` should only be checked for v2 tables.  If we use the parameterized `formatVersion`, then there will introduce more unit tests that be ignored because we will usually provide a line in `testV2RemoveMetaColumn` as :  Assume.assumeTrue("xxx", formatVersion==2).

Reply:
This make sense.